soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -371,8 +377,83 @@ namespace Logic {
   bool BuyDekuShieldPast        = false;
   bool TimeTravelPast           = false;
 
-  bool CanPlay(bool song) {
-    return Ocarina && song;
+  bool CanPlay(bool hasSong, Song song) {
```

Comment:
I think it would be best to merge this function into CanUse, where you only need to supply the itemName. CanUse is coded in more concise and readable way and using the itemName saves you needing a custom enum. Keys.hpp is being broken up in rando v3, but that's something for rando v3 for handle not a PR pointing at develop.

Reply:
Done